projects/polynomial/index.js

```diff
@@ -83,16 +101,85 @@ async function tvl (timestamp, ethBlock, chainBlocks) {
   tokens_balances_pairs.forEach( e => {
     let mainnetTokenAddress = getL1SynthAddressFromL2SynthAddress(e[0]);
     if( balances.hasOwnProperty(mainnetTokenAddress) ){
-      balances[mainnetTokenAddress] = BigNumber(balances[mainnetTokenAddress]).plus( BigNumber(e[1]) ).toFixed(0);
+      balances[mainnetTokenAddress] = BigNumber(balances[mainnetTokenAddress]).plus( BigNumber(e[1]) )
     } else{
       balances[mainnetTokenAddress] = e[1];
     }
   })
+  const v2balance = await tvl2(block)
+
+  Object.keys(balances).forEach(key => {
+    const v1Amount = balances[key]
+    const v2Amount = v2balance[key]
+
+    if (v2Amount) {
+      balances[key] = BigNumber(v1Amount).plus(v2Amount)
+    }
+  })
 
   return balances
 }
 
 
+function tvlSummup(contract) {
+  const { totalFund, withdrawal, deposit } = contract
+
+  return new BigNumber(totalFund)
+    .plus(new BigNumber(deposit))
+    .minus(new BigNumber(withdrawal))
+}
+async function tvl2(block) {
+  const ENDPOINT = 'https://earn-api.polynomial.fi/vaults'
```

Comment:
is there a way to look up vaut list on chain?

Reply:
We can't get vault list on-chain.